SHiELD/coupler_main.F90

```diff
@@ -0,0 +1,464 @@
+
+   namelist /coupler_nml/ current_date, calendar, force_date_from_namelist, &
+                          months, days, hours, minutes, seconds, iau_offset,  &
+                          dt_atmos, dt_ocean, atmos_nthreads, memuse_verbose, &
+                          use_hyper_thread, restart_secs, restart_days, &
+                          restart_start_secs, restart_start_days, &
+                          restart_secs_aux, restart_days_aux, &
+                          restart_start_secs_aux, restart_start_days_aux, &
+                          restart_duration_secs_aux, restart_duration_days_aux
+
+! ----- local variables -----
+   character(len=32) :: timestamp
+   logical :: intrm_rst, intrm_rst_1step
+
+!#######################################################################
+
+ call fms_init()
+ call sat_vapor_pres_init()
+ call constants_init()
+
+ initClock = mpp_clock_id( 'Initialization' )
+ call mpp_clock_begin (initClock) !nesting problem
+
+ call coupler_init
+ call print_memuse_stats('after coupler init')
+
+ call mpp_set_current_pelist()
+ call mpp_clock_end (initClock) !end initialization
+ mainClock = mpp_clock_id( 'Main loop' )
+ termClock = mpp_clock_id( 'Termination' )
+ call mpp_clock_begin(mainClock) !begin main loop
+
+ do nc = 1, num_cpld_calls
+
+    Time_atmos = Time_atmos + Time_step_atmos
+
+    call update_atmos_model_dynamics (Atm)
+
+    call update_atmos_radiation_physics (Atm)
+
+    call update_atmos_model_state (Atm)
+
+!--- intermediate restart
+    if (intrm_rst) then
+      if (nc /= num_cpld_calls) then
+        if (intrm_rst_1step .and. nc == 1) then
+          timestamp = date_to_string (Time_atmos)
+          call atmos_model_restart(Atm, timestamp)
+          call coupler_restart(timestamp)
+        endif
+        if (Time_atmos == Time_restart .or. Time_atmos == Time_restart_aux) then
+          if (Time_atmos == Time_restart) then
+            timestamp = date_to_string (Time_restart)
+          else
+            timestamp = date_to_string (Time_restart_aux)
+          endif
+          call atmos_model_restart(Atm, timestamp)
+          call coupler_restart(timestamp)
+          if (Time_atmos == Time_restart) &
+              Time_restart = Time_restart + Time_step_restart
+          if ((restart_secs_aux > 0 .or. restart_days_aux > 0) .and. &
+              Time_atmos == Time_restart_aux .and. &
+              Time_restart_aux < Time_restart_end_aux) then
+            Time_restart_aux = Time_restart_aux + Time_step_restart_aux
+          endif
+        endif
+      endif
+    endif
+
+    call print_memuse_stats('after full step')
+
+ enddo
+
+!-----------------------------------------------------------------------
+
+ call mpp_set_current_pelist()
+ call mpp_clock_end(mainClock)
+ call mpp_clock_begin(termClock)
+
+ call coupler_end
+ call mpp_set_current_pelist()
+ call mpp_clock_end(termClock)
+
+ call fms_end
+
+!-----------------------------------------------------------------------
+
+ stop
```

Comment:
Does this need to be in here?  fms_end should end the program.  THe other coupler_main files end with fms_end.